workers/unity/Packages/com.improbable.gdk.core/Collections/MessageList.cs

```diff
@@ -45,6 +45,39 @@ public void Add(in T item)
             ++Count;
         }
 
+        public void Insert(int index, in T item)
+        {
+            if (items.Length <= Count)
+            {
+                int targetLength = items.Length == 0 ? 4 : items.Length * 2;
+                var temp = new T[targetLength];
+                Array.Copy(items, 0, temp, 0, index);
+                temp[index] = item;
+                Array.Copy(items, index, temp, index + 1, Count - index);
+                items = temp;
+            }
+            else
+            {
+                Array.Copy(items, index, items, index + 1, Count - index);
+                items[index] = item;
+            }
+
+            ++Count;
+        }
+
+        public void InsertSorted(in T item, IComparer<T> comparer)
+        {
+            var index = Array.BinarySearch(items, 0, Count, item, comparer);
```

Comment:
The way this function works is excellent

Reply:
> The index of the specified value in the specified array, if value is found; otherwise, a negative number. If value is not found and value is less than one or more elements in array, the negative number returned is the bitwise complement of the index of the first element that is larger than value. If value is not found and value is greater than all elements in array, the negative number returned is the bitwise complement of (the index of the last element plus 1).

For those interested